provider/plugin/plugin.go

```diff
@@ -0,0 +1,337 @@
+// `{propertyvaluesequal: true}`
+// Errors in anything technically happening from the provider will return true so that no update is performed.
+// Errors will also be logged and exposed as metrics so that it is possible to alert on them if needed.
+func (p PluginProvider) PropertyValuesEqual(name string, previous string, current string) bool {
+	u, err := url.JoinPath(p.remoteServerURL.String(), "propertyvaluesequal")
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Debugf("error joining path: %s", err)
+		return true
+	}
+	b, err := json.Marshal(&PropertyValuesEqualRequest{
+		Name:     name,
+		Previous: previous,
+		Current:  current,
+	})
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Debugf("error marshaling request: %s", err)
+		return true
+	}
+
+	req, err := http.NewRequest("POST", u, bytes.NewBuffer(b))
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Debugf("error creating request: %s", err)
+		return true
+	}
+	resp, err := p.client.Do(req)
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Debugf("error performing request: %s", err)
+		return true
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Debugf("failed to apply changes with code %d", resp.StatusCode)
+		return true
+	}
+
+	respoBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Errorf("failed to read body: %v", err)
+		return true
+	}
+	r := PropertyValuesEqualResponse{}
+	err = json.Unmarshal(respoBody, &r)
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Errorf("failed to unmarshal body: %v", err)
+		return true
+	}
+	return r.Equals
+}
+
+// AdjustEndpoints will call the provider doing a POST on `/adjustendpoints` which will return a list of modified endpoints
+// based on a provider specific requirement.
+// This method returns an empty slice in case there is a technical error on the provider's side so that no endpoints will be considered.
+func (p PluginProvider) AdjustEndpoints(e []*endpoint.Endpoint) []*endpoint.Endpoint {
+	endpoints := []*endpoint.Endpoint{}
+	u, err := url.JoinPath(p.remoteServerURL.String(), "adjustendpoints")
+	if err != nil {
+		adjustEndpointsErrorsGauge.Inc()
+		log.Debugf("error joining path, %s", err)
+		return endpoints
+	}
+	b, err := json.Marshal(e)
+	if err != nil {
+		adjustEndpointsErrorsGauge.Inc()
+		log.Debugf("error marshaling endpoints, %s", err)
+		return endpoints
+	}
+	req, err := http.NewRequest("POST", u, bytes.NewBuffer(b))
+	if err != nil {
+		adjustEndpointsErrorsGauge.Inc()
+		log.Debugf("error creating new HTTP request, %s", err)
+		return endpoints
+	}
+	resp, err := p.client.Do(req)
+	if err != nil {
+		adjustEndpointsErrorsGauge.Inc()
+		log.Debugf("error in http request, %s", err)
+		return endpoints
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
```

Comment:
I think the docs say StatusCode 20x is fine, but here it's only 200, so we should either change docs or what we expect here.

Reply:
Switched the docs to 200.